app.js

```diff
@@ -12,11 +12,22 @@ const data = {
   vegetables: []
 }
 
+
 app.get('/vegetables', (req, res, next) => {
   const { vegetables } = data
-  res.json(vegetables)
+  const { name } = req.query
+  const vegetable = vegetables.filter(veggie => veggie.name.includes(name))
+
+  if (!name) {
+    res.json(vegetables)
+  } else {
+    res.status(200).json(vegetable)
+  }
+  res.json({ status: 200, response: veggie })
```

Comment:
In this case, you'll be responding twice on any request. This is causing errors such as:
```
Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
```